File: pattern/singleton/singleton.cpp
#include<iostream>
 
using namespace std;

class SingletonClass
{
private:
    static SingletonClass instance; // 정적 변수, 정적 싱글톤
    SingletonClass() {
       
    }
    
public:
     ~SingletonClass() {

     }
    static SingletonClass& getInstance() {
        return instance;
    }
    void ShowMessage() {
        cout << "싱글톤 패톤" << endl;
    }
};

SingletonClass SingletonClass::instance;

int main() {
    SingletonClass::getInstance().ShowMessage();
    
    return 0;
}

File: C-sharp/code/singleton.cs
// 싱글톤
using System;

namespace MyApp
{
    
    public class Singleton
    {
        private static Singleton instance;
        public static Singleton getInstance()
        {
            if (instance == null)
            {
                instance = new Singleton();
            }
            return instance;
        }

        public void ShowMessage()
        {
            Console.WriteLine("싱글톤 패턴");
        }

    }
    internal class Program
    {
        static void Main(string[] args)
        {
            // 클래스로 바로 접근 가능한 static 함수를 통해 
            Singleton.getInstance().ShowMessage();

        }
    }
}